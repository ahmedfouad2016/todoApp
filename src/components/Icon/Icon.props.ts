import {ImageStyle, StyleProp, ViewStyle} from 'react-native';
import {icons} from 'utils/icons';

export type IconTypes = keyof typeof icons;

export interface IconProps {
  style?: StyleProp<ImageStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  icon: IconTypes;
  size?: number;
}
