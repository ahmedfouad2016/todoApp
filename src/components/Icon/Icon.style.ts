import {I18nManager, StyleSheet} from 'react-native';

export default StyleSheet.create({
  image: {
    resizeMode: 'contain',
    transform: [{scaleX: I18nManager.isRTL ? -1 : 1}],
  },
});
