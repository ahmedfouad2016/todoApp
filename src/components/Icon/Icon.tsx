import React from 'react';
import {View, Image, StyleSheet} from 'react-native';
import {IconProps} from './Icon.props';
import {icons} from 'utils/icons';
import styles from './Icon.style';

export function Icon(props: IconProps) {
  const {style: styleOverride, icon, containerStyle, size} = props;
  const style = StyleSheet.flatten([
    styles.image,
    {width: size, height: size},
    styleOverride,
  ]);
  return (
    <View style={containerStyle}>
      <Image style={style} source={icons[icon]} />
    </View>
  );
}
