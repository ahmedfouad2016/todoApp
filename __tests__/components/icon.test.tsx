/**
 * @format
 */

import 'react-native';
import React from 'react';
import {Icon} from '@components/Icon';

import {render} from '@testing-library/react-native';

describe('Icon Component', () => {
  it('should Icon render ', () => {
    const wrapper = render(<Icon icon="add" size={25} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should Icon size reflect on width and height', () => {
    const result = {height: 25, width: 25};
    const {getByTestId} = render(<Icon icon="add" size={25} />);
    const img = getByTestId('icon-image');
    expect(img.props.style).toMatchObject(result);
  });
});
