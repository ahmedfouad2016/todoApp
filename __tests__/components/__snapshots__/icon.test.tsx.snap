// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component should Icon render  1`] = `
<View>
  <RCTImageView
    internal_analyticTag={null}
    resizeMode="contain"
    source={
      Array [
        Object {
          "testUri": "../../../src/assets/icons/add.png",
        },
      ]
    }
    style={
      Object {
        "height": 25,
        "overflow": "hidden",
        "resizeMode": "contain",
        "width": 25,
      }
    }
    testID="icon-image"
  />
</View>
`;
